src/utility.hpp

```diff
@@ -821,6 +835,40 @@ string file_base_name(const string& filename);
 /// Only works for files readable by the current user.
 bool file_exists(const string& filename);
 
+/// Determine if a file can be written to.
+/// Only works for files writable by the current user.
+/// Meant to be used to validate output file names.
+/// Will actually open the file, so it will overwrite the current contents.
+bool file_can_be_written(const string& filename);
+
+/// Check if a file exists and return its name (if so) or error.
+/// Uses file_exists() and is intended to be called when parsing arguments.
+string error_if_file_does_not_exist(const string& context, const string& filename);
+
+/// Error if a file looks like it's gzipped.
+void error_if_file_is_gzipped(const string& context, const string& filename);
+
+/// Check if a file can be written to and return its name (if so) or error.
+/// Uses file_can_be_written() and is intended to be called when parsing arguments.
+string error_if_file_cannot_be_written(const string& context, const string& filename);
```

Comment:
How do these various file-checking functions interact with "-", which we can use in many places as representing "read from standard input"? (For example, I'm pretty sure you can `vg filter --exact-name -n MyRead reads.gam | vg view -dp graph.vg -A - | dot -Tsvg -o graph_with_read.svg` in order to have `vg view` read a graph from a file and a read to visualize on it from standard input.

Probably we want "-" to *not* appear writable, even though you probably could create and write to `./-`. And we would want it to appear as if it *does* exist, because if we use any of the `get_input_file()` functions to open it, it effectively exists.

But then, are the checkers always used with those accessors that support "-"?